### DISCUSSION TITLE
请问alistv3使用onlyoffice，如何填写url

### DISCUSSION THREAD

**Discussion by hi778899** [AlistGo/alist#1731] [Q&A]
![image](https://user-images.githubusercontent.com/88808769/191256833-83a4d951-38f8-4ac9-8e8d-b05a191c51d6.png)
请问you_only_office_url如何填写？
我填写后会直接转跳到onlyoffice的welcome页面

**Comment by anwen-anyi** [comment]
应该是这样写吧？第三行，主要是加粗部分


{
	"doc,docx,xls,xlsx,ppt,pptx": {
		"Microsoft":"https://view.officeapps.live.com/op/view.aspx?src=$e_url",
		"Google":"https://docs.google.com/gview?url=$e_url&embedded=true",
		"onlyoffice":"**https://onlyoffice_URL**/view.html?src=$e_url"
	},

**Reply by anwen-anyi** [reply]
但是这样写，打开onedrive的没问题，但是阿里或者其他的就不行....
目前 OneDriver，Teambition，139个人（家庭没测试）这三个可以，
阿里 天翼 不知道是设置的不对还是哪里问题这样填写打不开....

**Reply by hi778899** [reply]
alist的onlyoffice的接口文件不知道咋写

**Reply by anwen-anyi** [reply]
> alist的onlyoffice的接口文件不知道咋写

你搭建好 **onlyoffice** 了吗？ 如果搭建好了你看我不是特意把URL加粗了吗你就把加粗的部分改一下后面不用变 
或者你私发我域名我帮你试试看

**Reply by zhushen12580** [reply]
> > alist的onlyoffice的接口文件不知道咋写
> 
> 你搭建好 **onlyoffice** 了吗？ 如果搭建好了你看我不是特意把URL加粗了吗你就把加粗的部分改一下后面不用变 或者你私发我域名我帮你试试看

大佬 helpe me
onlyoffice_URL在哪里找到呀？

**Reply by Masterlovsky** [reply]
> > > alist的onlyoffice的接口文件不知道咋写
> > 
> > 
> > 你搭建好 **onlyoffice** 了吗？ 如果搭建好了你看我不是特意把URL加粗了吗你就把加粗的部分改一下后面不用变 或者你私发我域名我帮你试试看
> 
> 大佬 helpe me onlyoffice_URL在哪里找到呀？

就是你自己的onlyoffice服务的url，可以是域名也可以是ip+端口号，view.html可能需要自己写。你可以看一下discussion#3255

**Reply by zhushen12580** [reply]
> > > > alist的onlyoffice的接口文件不知道咋写
> > > 
> > > 
> > > 你搭建好 **onlyoffice** 了吗？ 如果搭建好了你看我不是特意把URL加粗了吗你就把加粗的部分改一下后面不用变 或者你私发我域名我帮你试试看
> > 
> > 
> > 大佬 helpe me onlyoffice_URL在哪里找到呀？
> 
> 就是你自己的onlyoffice服务的url，可以是域名也可以是ip+端口号，view.html可能需要自己写。你可以看一下discussion#3255

好的，谢谢，我先琢磨琢磨。

**Reply by nxliubo** [reply]
做成了，分享一下呗？！

**Reply by CrazyJasonwell** [reply]
大神，我装了onlyoffice，需要在iframe中配置https://onlyoffice_URL/view.html?src=$e_url，看了一些帖子,但是不知道将view.html放到onlyoffice的哪个目录里，才能实现文件预览？清大神指点一下。

**Reply by Masterlovsky** [reply]
可以和api.js放在一个目录下

**Comment by zhushen12580** [comment]
大佬 view.html文件放到哪个文件夹呀？











只要你有梦想，全世界都为你让路。

---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2023年3月9日(周四) 晚上10:55
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [alist-org/alist] 请问alistv3使用onlyoffice，如何填写url (Discussion #1731)




    
alist的onlyoffice的接口文件不知道咋写
  
你搭建好 onlyoffice 了吗？ 如果搭建好了你看我不是特意把URL加粗了吗你就把加粗的部分改一下后面不用变 或者你私发我域名我帮你试试看
  
大佬 helpe me onlyoffice_URL在哪里找到呀？
  
就是你自己的onlyoffice服务的url，可以是域名也可以是ip+端口号，view.html可能需要自己写。你可以看一下discussion#3255

**Reply by Masterlovsky** [reply]
可以放api.js旁边儿，参考一下 https://github.com/alist-org/alist/discussions/3255

**Reply by zhushen12580** [reply]
> 可以放api.js旁边儿，参考一下 #3255

大佬遇到新问题了。
可以访问view.html文件，加了iframe配置但是还是报这个“The "document.fileType" parameter for the config object is invalid. Please correct it.”

alist:v3.13.2
onlyoffice:windows社区版

**Reply by Masterlovsky** [reply]
能访问到api.js吗，view里面的js地址改了吗?iframe里面的配置发一下

**Reply by zhushen12580** [reply]
> 能访问到api.js吗，view里面的js地址改了吗?iframe里面的配置发一下

可以访问，js地址也改了
`{
"doc,docx,xls,xlsx,ppt,pptx": {
"onlyoffice":"http://yunpan.zhushen12580.top:9000/web-apps/apps/api/documents/view.html?src=$e_url",
"Microsoft":"https://view.officeapps.live.com/op/view.aspx?src=$e_url",
"Google":"https://docs.google.com/gview?url=$e_url&embedded=true"
},
"pdf": {
"PDF.js":"https://alist-org.github.io/pdf.js/web/viewer.html?file=$e_url"
},
"epub": {
"EPUB.js":"https://alist-org.github.io/static/epub.js/viewer.html?url=$e_url"
},
"drawio,vsdx,vdx,gliffy,lucid":{
"draw.io":"https://viewer.diagrams.net/index.html?tags={}&target=self&highlight=0000ff&edit=_blank&layers=1&nav=1&title=$name#U$durl"
}
}`

**Reply by Masterlovsky** [reply]
你的onlyoffice的服务器或者容器可以连上你的alist吗，要双向互通才行，就是从onlyoffice站点curl你的alist也要能通。我不清楚是不是onlyoffice-windows社区版本的问题，我是docker安装的

**Reply by zhushen12580** [reply]
> 你的onlyoffice的服务器或者容器可以连上你的alist吗，要双向互通才行，就是从onlyoffice站点curl你的alist也要能通。我不清楚是不是onlyoffice-windows社区版本的问题，我是docker安装的

问题发现了iframe的配置应该是$durl而非$e_url

**Reply by zhushen12580** [reply]
> 你的onlyoffice的服务器或者容器可以连上你的alist吗，要双向互通才行，就是从onlyoffice站点curl你的alist也要能通。我不清楚是不是onlyoffice-windows社区版本的问题，我是docker安装的

大佬 您知道怎么关闭onlyoffice的jwt吗

**Reply by anwen-anyi** [reply]
> > 你的onlyoffice的服务器或者容器可以连上你的alist吗，要双向互通才行，就是从onlyoffice站点curl你的alist也要能通。我不清楚是不是onlyoffice-windows社区版本的问题，我是docker安装的
> 
> 
> 
> 问题发现了iframe的配置应该是$durl而非$e_url

😂我发的时候我记得还没有 durl这个选项似乎

**Reply by ganjiatao** [reply]
> > 你的onlyoffice的服务器或者容器可以连上你的alist吗，要双向互通才行，就是从onlyoffice站点curl你的alist也要能通。我不清楚是不是onlyoffice-windows社区版本的问题，我是docker安装的
> 
> 大佬 您知道怎么关闭onlyoffice的jwt吗
兄弟，你解决了jwt的问题没有

**Reply by Masterlovsky** [reply]
> > > 你的onlyoffice的服务器或者容器可以连上你的alist吗，要双向互通才行，就是从onlyoffice站点curl你的alist也要能通。我不清楚是不是onlyoffice-windows社区版本的问题，我是docker安装的
> > 
> > 
> > 大佬 您知道怎么关闭onlyoffice的jwt吗
> > 兄弟，你解决了jwt的问题没有

我是直接安装了7.11版本的onlyoffice，7.11之前的版本不会强制开启jwt。ps：贴一个ChatGPT的回答，如果不管用不要怪我，我没试过，你可以尝试一下。
------------------------
Answered by ChatGPT:
要关闭onlyoffice的jwt认证功能，您需要在onlyoffice配置文件中做出以下更改：
1. 打开onlyoffice的配置文件/etc/onlyoffice/documentserver/default.json。
2. 找到`"token":": }这一行。
3. 将"enable": true改为"enable": false。
4. 保存并关闭文件。
重启onlyoffice服务，以便使配置文件生效认证之后，onlyoffice将不再要求用户进行登录，并且所有文档都可以在匿名用户状态下进行编辑。但是，请注意，这也意味着任何人都可以访问onlyoffice服务并对文档进行更改。如果不想让所有人具有这样的权限，请考虑配置其他安全机制来保护您的文档。

**Reply by ganjiatao** [reply]
> > > > 你的onlyoffice的服务器或者容器可以连上你的alist吗，要双向互通才行，就是从onlyoffice站点curl你的alist也要能通。我不清楚是不是onlyoffice-windows社区版本的问题，我是docker安装的
> > > 
> > > 
> > > 大佬 您知道怎么关闭onlyoffice的jwt吗
> > > 兄弟，你解决了jwt的问题没有
> 
> ## 我是直接安装了7.11版本的onlyoffice，7.11之前的版本不会强制开启jwt。ps：贴一个ChatGPT的回答，如果不管用不要怪我，我没试过，你可以尝试一下。
> Answered by ChatGPT: 要关闭onlyoffice的jwt认证功能，您需要在onlyoffice配置文件中做出以下更改：
> 
> 1. 打开onlyoffice的配置文件/etc/onlyoffice/documentserver/default.json。
> 2. 找到`"token":": }这一行。
> 3. 将"enable": true改为"enable": false。
> 4. 保存并关闭文件。
>    重启onlyoffice服务，以便使配置文件生效认证之后，onlyoffice将不再要求用户进行登录，并且所有文档都可以在匿名用户状态下进行编辑。但是，请注意，这也意味着任何人都可以访问onlyoffice服务并对文档进行更改。如果不想让所有人具有这样的权限，请考虑配置其他安全机制来保护您的文档。

谢谢兄弟

**Comment by zhushen12580** [comment]
安稳大佬 您onlyoffice的edit.html可以发一份吗











只要你有梦想，全世界都为你让路。

---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2023年3月12日(周日) 晚上11:53
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [alist-org/alist] 请问alistv3使用onlyoffice，如何填写url (Discussion #1731)




   
你的onlyoffice的服务器或者容器可以连上你的alist吗，要双向互通才行，就是从onlyoffice站点curl你的alist也要能通。我不清楚是不是onlyoffice-windows社区版本的问题，我是docker安装的
  
问题发现了iframe的配置应该是$durl而非$e_url
  
😂我发的时候我记得还没有 durl这个选项似乎

**Reply by anwen-anyi** [reply]
我没有啊，我用的是别人的 onlyoffice

**Comment by quhanj** [comment]
借楼一问，这个是改哪个文件啊？
